SSNFSd/ssnfsworker.cpp

```diff
@@ -140,6 +140,32 @@ void SSNFSWorker::run()
     quit();
 }
 
+bool SSNFSWorker::isValidPath(QString path){
+    // Checking if it begins with /
+    if (!path.startsWith('/')){
+        return false;
+    }
+    if (path.endsWith('/')){
+        path.truncate(path.length()-1);
+    }
+
+    QStringList directories = path.split("/");
+
+    qint8 counter = 0;
+
+    for (int i = 1; i < directories.size(); ++i){
+            if (directories.at(i) == ".."){
+                 counter--;
+                if (counter < 0) return false;
```

Comment:
Note that if the path starts with a `/`, the first element will be an empty string. Therefore, an invalid path like `/../Something` will return true.
I see a few possible solutions. The simplest of which is to either remove the first `/` from the string or remove the first element from the QStringList.

Reply:
Yea, that is why I started the loop from i = 1. I tested it with the `/../Something` and it returns false.
Should I still make the changes ?

Reply:
I think that you should still make the change because `/dir//////file` = `/dir/file` and `////../file` = `/../file`.
In other words, empty parts don't change the directory, therefore the counter should not be changed. In fact, now that I think about it, parts that contain just `.` also need to be disregarded, since in that case the directory also doesn't change.

You could consider simply changing the `else` component to `else if` and make sure the string is not empty and is not `.` before incrementing the counter.

Reply:
I have made the changes